### PR TITLE
Fixes #17767: Do a `git gc` periodically

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/17767

### CODE REVIEW

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/batch/GitGC.scala [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    def isCancelled = false
+  }
+
+  // must not fail, will be in a cron
+  val gitgc: UIO[Unit] = for {
+    t0 <- currentTimeMillis
+    _  <- gitRepo.semaphore.withPermit(IOResult.effect {
+           gitRepo.git.gc().setProgressMonitor(new LogProgressMonitor()).call
+         }).catchAll(err => logger.error(s"Error when performing GitGC on ${gitRepo.rootDirectory.name}: ${err.fullMsg}"))
```
Comment (amousset) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/batch/GitGC.scala:93]: we should decide between "gitgc", "GitGC" in logs. I would prefer `git-gc`, but anything goes.

File: webapp/sources/rudder/rudder-web/src/main/resources/configuration.properties.sample [hunk 2]
```diff
@@ -410,6 +410,21 @@ rudder.dir.gitRoot=/var/rudder/configuration-repository
...
+# Periodically do a Git GC to avoid accumulation of garbage.
+# The schedule is defined based on a cron-like format with the usual fields plus a second one:
+# seconds minutes hourOfDay dayOfTheMonth month DayOfTheWeek
+# See https://www.alonsodomin.me/cron4s/userguide/#parsing for more information, and
+# https://en.wikipedia.org/wiki/Cron for even more information.
+#
+# You can disable it totally with:
+#   @never
+# in place of the cron usual format.
+# Even on big systems, once a day should be enought.
```
Comment (amousset) [inline webapp/sources/rudder/rudder-web/src/main/resources/configuration.properties.sample:423]:
```suggestion
# Even on big systems, once a day should be enough.
```

File: webapp/sources/rudder/rudder-web/src/main/resources/configuration.properties.sample [hunk 3]
```diff
@@ -410,6 +410,21 @@ rudder.dir.gitRoot=/var/rudder/configuration-repository
 #
 rudder.config.repo.new.file.group.owner=rudder
 
+#
+# Periodically do a Git GC to avoid accumulation of garbage.
+# The schedule is defined based on a cron-like format with the usual fields plus a second one:
```
Comment (amousset) [inline webapp/sources/rudder/rudder-web/src/main/resources/configuration.properties.sample:415]: "a second one" is ambiguous here, maybe  "with the usual fields, preceded by a field for seconds".

File: webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderConfig.scala [hunk 4]
```diff
@@ -387,6 +389,18 @@ object RudderConfig extends Loggable {
     }
   }
 
+  val RUDDER_GIT_GC = (try {
+    config.getString("rudder.git.gc")
+  } catch {
+    // missing key, perhaps dû to migration, use default
```
Comment (amousset) [inline webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderConfig.scala:395]:
```suggestion
    // missing key, perhaps due to migration, use default
```

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**